src/steps/command/plugins.ts

```diff
@@ -12,6 +12,8 @@ export class Plugin {
 
 export interface Plugins<T> {
     add(plugin: Plugin): T;
+    /** Return a list of plugins that match the given predicate */
+    filter: Plugin[]['filter'];
```

Comment:
Nifty but I think readability > DRY in this case

```suggestion
    filter(predicate: (plugin: Plugin) => boolean): Plugin[];
```

Reply:
disagree. We're directly passing through the call to the array and the simplified version above is a subset of what `filter` takes as a param. You can see in an older version that it was like what you suggested before.

Reply:
No strong opinions but like I said earlier, I think de-duping has less utility having 1 less layer of indirection. Plus it has the teeny tiny advantage of not forcing `T` to store the plugins in an array